wpilibc/src/main/native/include/frc/Tracer.h

```diff
@@ -16,28 +25,225 @@ class raw_ostream;
 
 namespace frc {
 /**
- * A class for keeping track of how much time it takes for different parts of
- * code to execute. This is done with epochs, that are added to calls to
- * AddEpoch() and can be printed with a call to PrintEpochs().
- *
- * Epochs are a way to partition the time elapsed so that when overruns occur,
- * one can determine which parts of an operation consumed the most time.
+ * A Utility class for tracing code execution time. Will put info to
+ * NetworkTables under the "Tracer" table. The times outputted by the {@link
+ * Tracer} are in milliseconds.
```

Comment:
```suggestion
 * Tracer} are in milliseconds.
 * <p>Example inside {@code Robot.cpp}
 *
 * <pre><code>
 *
 * void Robot::RobotPeriodic() {
 *   frc2::CommandScheduler::GetInstance().Run();
 *   frc::Tracer::traceFunc("MyVendorDep", MyVendorDep::updateAll);
 * }
 * </code></pre>
 *
 * <p>Example inside a {@code Drive Subsystem}
 *
 * <pre><code>
 * // Subsystem periodics are automatically traced
 * void DriveSubsystem::Periodic() {
 *   for (auto module : modules) {
 *     frc::Tracer::traceFunc("Module" + module.getName(), [this] { module.update(); });
 *   }
 *   frc::Tracer::traceFunc("Gyro", [this] { gyro.update(); });
 * }
 * </code></pre>
```

Reply:
`traceFunc` should be `TraceFunc` and `MyVendorDep::updateAll` should be `[] { MyVendorDep::UpdateAll(); }`, but yep, that would work!